### PR TITLE
Consider command line argument for output directory

### PR DESCRIPTION
**Pull request by weliveindetail**
Zurzeit wird der -o Parameter ja nicht beachtet.. kleiner Fix dafür, geht aber bestimmt noch schöner. Ich bin glaub nicht so gut mit C und Strings... ^^

### CODE REVIEW

File: src/protog.cpp [hunk 1]
```diff
@@ -36,6 +39,21 @@ void print_help(FILE* f) {
     fprintf(f, "  protog -p openrtb.proto -m com.google.openrtb.BidRequest -i openrtb.pb.h\n");
 }
 
+void make_output_dir(const char* output_dir, char* buffer) {
+    struct stat st = {0};
+    if (stat(output_dir, &st) == -1) {
+        mkdir(output_dir, 0700);
```
Comment (0x7f) [inline src/protog.cpp]: Can lead to a race condition. Better just call `mkdir()` and check for `EEXIST`.

File: src/yajl_writer.h [hunk 2]
```diff
@@ -8,18 +8,18 @@ namespace protog {
...
 
-    virtual void write(const Graph &graph, const char* proto_header) override {
+    virtual void write(const Graph &graph, const char* proto_header, const char* output_dir) override {
         auto name_lower = graph.root.desc->name();
         std::transform(name_lower.begin(), name_lower.end(), name_lower.begin(), ::tolower);
         const auto cpp_type = get_full_cpp_type_name(*graph.root.desc);
         const auto res_name_prefix = name_lower + "_parser.pb";
 
-        const auto header_name = res_name_prefix + ".h";
+        const auto header_name = output_dir + res_name_prefix + ".h";
```
Comment (0x7f) [inline src/yajl_writer.h:17]: Too bad the [filesystem api](http://en.cppreference.com/w/cpp/experimental/fs/path/append) is not widely supported yet...

Comment (0x7f) [inline src/yajl_writer.h:17]: Also did not want to pull in `boost::filesystem` just for that feature.

**Comment by 0x7f** [comment]
Hehe, thanks for the patch!

File: src/protog.cpp (hunk 1 above)
Comment (weliveindetail) [inline src/protog.cpp]: Hmm.. and if it exists we'd need to delete the files that we'll write to? All those little details ^^

File: src/yajl_writer.h (hunk 2 above)
Comment (weliveindetail) [inline src/yajl_writer.h:17]: Yap that's probably not worth it

**Pull request closed without merging**